internal/identity/identity.go

```diff
@@ -0,0 +1,199 @@
+package identity
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/alecthomas/types/optional"
+	"google.golang.org/protobuf/proto"
+
+	v1 "github.com/TBD54566975/ftl/backend/protos/xyz/block/ftl/v1"
+	"github.com/TBD54566975/ftl/internal/model"
+)
+
+type Identity interface {
+	String() string
+}
+
+var _ Identity = Controller{}
+
+type Controller struct{}
+
+func NewController() Controller {
+	return Controller{}
+}
+
+func (c Controller) String() string {
+	return "ca"
+}
+
+var _ Identity = Runner{}
+
+// Runner identity
+// TODO: Maybe use KeyType[T any, TP keyPayloadConstraint[T]]?
+type Runner struct {
+	Key    model.RunnerKey
+	Module string
+}
+
+func NewRunner(key model.RunnerKey, module string) Runner {
+	return Runner{
+		Key:    key,
+		Module: module,
+	}
+}
+
+func (r Runner) String() string {
+	return fmt.Sprintf("%s:%s", r.Key, r.Module)
+}
+
+func Parse(s string) (Identity, error) {
+	if s == "" {
+		return nil, fmt.Errorf("empty identity")
+	}
+	parts := strings.Split(s, ":")
+
+	if parts[0] == "ca" {
```

Comment:
Might be good to have:
```suggestion
	if parts[0] == "ca" && len(parts) == 1 {
```